### ISSUE TITLE
Potentional for Duplicated or Missing Entries

### ISSUE THREAD

**Issue by dstufft** [twisted/newsbuilder#27]
As much as I can tell, newsbuilder will only reliably work in the case if you have a single integration branch from which you periodically cut releases or if you only ever apply patches directly to integration branches and never merge between integration branches. If you have multiple integration branches and you ever merge between them then you're at risk of either getting duplicated or missing change log items. It may be that these use cases are out of scope for newsbuilder, I wanted to file an issue though because at the very least it'd probably be good to document the limitations.
#### Duplicated Entries

If you have a monotonically increasing version number (such as pip does with 1.5.0, 1.5.1, ...) but you have two (or more) integration branches, one for the "current" release (e.g. pip's 1.5.x branch) and one for a "development" release (e.g. pip's develop branch) then you can run into a situation where you have duplicated entries. For instance, if a change is made to the 1.5.x branch then the 1.5.x branch is merged into develop (without a release being made) then the topfile will exist in both the future 1.6 and the next 1.5.x release. This will give the same entry to the next 1.5.x release, and the first 1.6.x release even though logically the releases are monotonically increasing and it shouldn't exist in the 1.6.x version.

Similarly the same thing will happen if you apply a patch to the develop branch and then cherry-pick the changes over to the 1.5.x release branch.
#### Missing Entries

If you maintain two release series in parallel (such as 2.x and 3.x) you likely want a change that lands in both series to show up in the changelog for both series. However to do this you have to (obviously) ensure that the topfile exists in all branches before the release is cut for each series. However this won't be the case if instead of applying the patch to both series you instead do a merge _after_ you've done the release process for the other series. This is due to the fact that the release process deletes the topfile, so it will no longer exist when you do the merge into the other release branch.
